tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -1291,6 +1291,37 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system"
 	},
 
+	/** Firmware layout for the EAP245 v3 */
+	{
+		.id     = "EAP245-V3",
```

Comment:
I've never seen this without .vendor defined, though empty. Does it still work and can all the empty vendor statements by dropped?

Reply:
TP-Link's firmware files for this device just contain the file length (minus the signature) and md5 checksum at the start of the file. Then it's FF until the partition list at 0x1014. This is true for firmware files for:
* EAP245 v1 and v3
* EAP225 v1, v2 and v3
If .vendor is not defined, then the produced factory image also contains FF after the md5 sum, replicating the TP-Link format.

As for empty-string .vendor definitions, I can't really tell if they would work. The Archer C60-v3's .vendor is emtpy, so the factory images have {00 00 00 00 00] after the checksum. TP-Link's latest image for this device however, contains a header of 0x28c bytes ("fw-type:Cloud\n[binary data]"). If these factory images currently work, then I would assume removing the empty .vendor strings wouldn't hurt, but I cannot test this as I don't have such a device.